setup.cfg

```diff
@@ -55,6 +55,10 @@ dev =
     tox
     build
     ruff
+    diff-cover
+    pyright
+    pyright_diff_quality_plugin @ git+https://github.com/dperl-dls/pyright_diff_quality_plugin.git
```

Comment:
Should: I really don't like us depending on individual's orgs like this. As in the issue can we get the code either into `diff-cover` or into the DLS org before we start depending on it?

Reply:
done. tidying it up so it has all the features to integrate into diff-cover will take a bit longer, I'd rather start using it sooner and work on that over time